### ISSUE TITLE
Error: 'data-path/gunicorn_config.py' doesn't exist - Ubuntu desktop 20.04

### ISSUE THREAD

**Issue by fopson** [MarkBaggett/domain_stats#24]
Hi,

I have been trying to test domain-stats on an ubuntu desktop desktop 20.04 before deploying it to a production server but to tool results in an error and I am not sure what I am doing wrong.

I installed the packages from github repo and went through setup steps following the documentation and first realized I couldn't call the tool by typing `domain_stats` as suggested by the documentation and this also reflected in the service file:
<img width="367" alt="domain-stats02" src="https://user-images.githubusercontent.com/46285552/114896450-41744d00-9e08-11eb-9430-fc63e851f74e.png">

Based on this output, I figured the name of the tool may have been updated to `domain-stats` at some point but maybe not yet reflected in the documentation:
<img width="272" alt="domain-stats01" src="https://user-images.githubusercontent.com/46285552/114894868-e9891680-9e06-11eb-805f-c501670cbd5f.png">

Next when trying to start the tool using `domain-stats`, I get the following `Error: 'data-path/gunicorn_config.py' doesn't exist` even though I can confirm the file was created during the setup process and does exist:
<img width="297" alt="domain-stats03" src="https://user-images.githubusercontent.com/46285552/114897924-9b294700-9e09-11eb-9a88-fe782f7be301.png">

I am running as root. Any idear as to why it won't work?

Also regarding running this in production. Am I right in assuming if I am running an instance of domain-stats, I don't need to run a seperate instance of freq_server since that feature is available in domain-stats?

**Comment by MarkBaggett** [comment]
Hi @fopson 

I see that I do have a few errors in the installation instructions.   Thank you.   

To start the program you must run the command domain-stats and pass it the path to a file containing your domain-stats database (and gunicorn_config.py file).   You create that database and gunicorn_config.py with the tool domain-stats-settings.   

There are many different installation options shown in the README including GITHUB and DOCKER based installations.  I am not sure which one exactly you followed but I will run through both of them and check them when I have a chance.  That may be  a few weeks out.   

I am confident that the steps I show recorded video shown on the web page work.  If a command shown in the README is different that the recorded video showing the successful download, configuration and launching is different.   Follow the video.

The service file would have to be manually adjusted to your environment.  The values that are preconfigured in there are specific to the Docker installation.

**Comment by MarkBaggett** [comment]
Upon closer inspection of your print screen, I have no idea what is going on.   Something appears to be misconfigured in your settings file.   You might delete the dmstats-data file and just recreate it.    I am suspicious of the trailing slash in your path as well.   You might try 'domain-stats ./dsstats-data'  instead of 'domain-stats ./dsstats-data/'

Mark

**Comment by fopson** [comment]
Thanks @MarkBaggett for looking into it. I am using ubuntu 20.04 and I have tried installing through the published pip package and also installing from the git repository, taken your advise and removed the trailing slash but unfortunately I get the same error when trying to start domain-stats in both cases. 

I am still trying to get it working in a lab and it will eventually be deployed in production once I have a working setup. For performance purposes, it will be deployed to a dedicated server which will serve half a douzen logstash nodes, that's why I am not using the virtual-env setup but rather installing it systemwide.

Below are detailed steps I followed in my attemps to install domainstats using the 2 different methods mentioned above. Hopefully mistakes can be cought if I am making any. In both case I am starting from a clean (freshly installed) VM.

### I - Install domain-stats from Published package

**1. Update system package list**
`sudo apt update`

**2. Update system**
`sudo apt upgrade`

**3. Check system version**
`lsb_release -a`

**4. Install prerequisites**
```
sudo apt-get install -y python3-pip git
sudo pip3 install --system python-whois
```
<img width="528" alt="dmstats1" src="https://user-images.githubusercontent.com/46285552/115620719-74797d80-a2ed-11eb-9e0b-004711d13c72.png">

**5. Install domain-stats published package**
`sudo python3 -m pip install --system domain-stats  `
<img width="535" alt="dmstats2" src="https://user-images.githubusercontent.com/46285552/115622180-6b89ab80-a2ef-11eb-835c-f6c8e79a8d27.png">
<img width="490" alt="dmstats3" src="https://user-images.githubusercontent.com/46285552/115622205-72b0b980-a2ef-11eb-9391-497e27445ac5.png">

**6. Create directory for domain-stats Data**
`mkdir ./dmstat-data`

**7. Check content of dmstat-data folder**
`ll  ./dmstat-data`
<img width="245" alt="dmstats4" src="https://user-images.githubusercontent.com/46285552/115622303-9247e200-a2ef-11eb-89f7-bc76621e6e63.png">


**8. Setup domain-stats**
`domain-stats-settings ./dmstat-data`
<img width="519" alt="dmstats5" src="https://user-images.githubusercontent.com/46285552/115622400-b277a100-a2ef-11eb-893e-25884a59614d.png">

**9. Check content of dmstat-data folder**
`ll  ./dmstat-data`
<img width="343" alt="dmstats6" src="https://user-images.githubusercontent.com/46285552/115622452-c58a7100-a2ef-11eb-86df-0d7b49a91ac6.png">

3 files were created as the result of the setup including a file named gunicorn_config.py

**10. Import top1m** 
`domain-stats-utils -i /usr/local/lib/python3.8/dist-packages/domain_stats/data/top1m.import -nx dmstat-data`
<img width="649" alt="dmstats7" src="https://user-images.githubusercontent.com/46285552/115622509-dd61f500-a2ef-11eb-80af-4fe5c54c1ca4.png">

**11. Check content of dmstat-data folder**
`ll  ./dmstat-data`
<img width="343" alt="dmstats8" src="https://user-images.githubusercontent.com/46285552/115622606-fbc7f080-a2ef-11eb-9898-bb37f180c529.png">

Numbered folders where created as the result of the import command

**12. Start domain-stats**
`domain-stats dmstat-data`
<img width="365" alt="dmstats9" src="https://user-images.githubusercontent.com/46285552/115622641-08e4df80-a2f0-11eb-8207-049a54bdc13c.png">

Domain-stats can find the config file but not the gunicorn_config.py.

**13. Double check content of dmstat-data folder**
`ll  ./dmstat-data`

**14. Check content of the generated config file**
`nano ./dmstat-data/domain_stats.yaml`
<img width="184" alt="dmstats11" src="https://user-images.githubusercontent.com/46285552/115624672-d8eb0b80-a2f2-11eb-9631-3cde3ea85fae.png">


**15. Try to start domain-stats again**
`domain-stats ./dmstat-data`
<img width="341" alt="dmstats10" src="https://user-images.githubusercontent.com/46285552/115622712-1dc17300-a2f0-11eb-9458-ada08df69c2f.png">

Same error.



### II - Install domain-stats from Git repo

**1. Update system package list**
`sudo apt update`

**2. Update system**
`sudo apt upgrade`

**3. Install prerequisites**
```
sudo apt-get install -y python3-pip git
sudo pip3 install --system python-whois
```

**4. Change directory to /opt and clone the repo**
```
cd /opt
sudo git clone https://github.com/markbaggett/domain_stats
cd domain_stats
```

**5. Install domain-stats systemwide**
`sudo python3 -m pip install --system .`

**6. Create directory for domain-stats Data**
`sudo mkdir ./dmstat-data`

**7. Become root (otherwise sub-folders in the /opt directory are not accessible)**
`sudo su`

**8. Setup domain-stats**
`domain-stats-settings ./dmstat-data`

**9. Check content of dmstat-data folder**
`ll  ./dmstat-data`

3 files were created as the result of the setup including a file named gunicorn_config.py

**10. Import top1m** 
`domain-stats-utils -i /usr/local/lib/python3.8/dist-packages/domain_stats/data/top1m.import -nx dmstat-data`

**11. Check content of dmstat-data folder**
`ll  ./dmstat-data`

Numbered folders where created as the result of the import command

**12. Start domain-stats**
`domain-stats dmstat-data`

Domain-stats can find the config file but not the gunicorn_config.py.

**13. Double check content of dmstat-data folder**
`ll  ./dmstat-data`

**14. Try to start domain-stats again**
`domain-stats ./dmstat-data`

Same error.

Hopefully this provides more information about my setup. I may be making an obvious mistake but I am not sure.

Any help will be welcomed.

Thanks

**Comment by MarkBaggett** [comment]
I don’t see any obvious mistakes.  I think you have given me what I need to be able to be able to reproduce the setup.   Let me look into this.  Give me a week or so.

Mark
________________________________

**Comment by MarkBaggett** [comment]
This was perfect!  I really appreciate you taking the time to document your install.  I was able to reproduce the problem.

A quick short term fix.   Your install will work if you give it the fully qualified path when you start domain-stats.    For example:

![image](https://user-images.githubusercontent.com/743693/115715687-63108f80-a346-11eb-8f13-8d56bae2e639.png)

I am not sure what the issue is yet, but I will get it fixed.

**Comment by fopson** [comment]
Thanks @MarkBaggett for the time to look into it. The short tem fix using the absolute path works.

**Comment by MarkBaggett** [comment]
I pushed a patch to my git repo that resolves this.   If you install domain stats from my repo like this it works. 

python3 -m pip install git+https://github.com/MarkBaggett/domain_stats 

I just have to package it up, give it a new version number and put it to pypi.org before you can pip install by name.  Then I’ll close the ticket.  

Mark

**Comment by fopson** [comment]
Thanks @MarkBaggett I really appreciate the time. Just one other thing (this might just be me by the way) I have been doing some testing and after getting mary errors (for all the ccTLDs), I was reading the RDAP vs ISC Support section and I think you are pointing to a link for more informations? If so, I think the hyperlink may be missing?
<img width="443" alt="dmstats12" src="https://user-images.githubusercontent.com/46285552/115925585-4de84d80-a479-11eb-98b7-c1701e6874af.png">
I just googled it for more information but I just though I would bring it to your attention in case you meant to insert a hyperlink there.

Thanks again for pointing me to a fix.

**Comment by MarkBaggett** [comment]
Ok.  Thanks.  I’ll look into that as well.   

I just pushed a new domain_stats to pypi.  It should be ok now.  I will do more more testing tomorrow and then close the ticket.  

Mark

**Comment by MarkBaggett** [comment]
I tested the package from pypi and it now works properly if you just 'pip install domain_stats'.

I also checked the link in the README to the RDAP status and the link appears to still be up and valid.  Here it is directly.  
https://www.icann.org/resources/pages/rdap-background-2018-08-31-en

I look forward to ccTLD resolution as well.   The ISC support for this has been delayed indefinitely.  I hope that will change moving forward.
